pkg/skaffold/deploy/kubectl.go

```diff
@@ -143,6 +169,8 @@ func (k *KubectlDeployer) readManifests(ctx context.Context, offline bool) (depl
 	if err != nil {
 		return nil, fmt.Errorf("listing manifests: %w", err)
 	}
+	// Clean the temporary directory that holds the manifests downloaded from GCS
+	defer os.RemoveAll(manifestTmpDir)
```

Comment:
move defer up to L143. go coding standards mentions to call defer immediately after object creation.

e.g. the way is written right now, if `util.ExpandPathsGlob` fails, this temporary dir wont be cleaned up.

Reply:
In your code, `downloadManifestsFromGCS` creates the temporary dir. 
With this approach, you have to clean up the temporary dir multiple places. 
I would suggest,  you change the signature of `downloadManifestsFromGCS` to 
`func downloadManifestsFromGCS(dest string, gcsManifests []string) error`
In this case, your flow would be
```
tempDir, err := ioutil.TempDir("", manifestsStagingFolder)
 if err != nil {			
  return fmt.Errorf("failed to create tmp directory: %w", err)	
}
defer os.RemoveAll(tempDir)
if err :=downloadManifestsFromGCS(tempDir, gcsManifests)  != nil {
	return nil, fmt.Errorf("downloading from GCS: %w", err)
}
l, err := util.ExpandPathsGlob(tmpDir, []string{"*"})
if err != nil {
  return nil, fmt.Errorf("expanding kubectl manifest paths: %w", err)
}
list = append(list, l...)
```

Reply:
I will use this comment to reply all the comments regarding the temporary directory. `func Dependencies` only list the manifests but the files were read and processed in `func readManifests`, which means cleanup has to happen in `readManifests`, which also means I can't use `ioutil.TempDir` to create a randomly named temp dir(since the deploy is synchronized, this should be safe ). 
You are right about
> e.g. the way is written right now, if util.ExpandPathsGlob fails, this temporary dir wont 
> be cleaned up.
I will move the defer to L170.

Reply:
sounds good.